stacks-signer/src/main.rs

```diff
@@ -0,0 +1,294 @@
+    };
+
+    let mut session = StackerDBSession::new(host.clone(), contract.clone());
+    session.connect(host, contract).unwrap();
+    let mut chunk_opt = session.get_chunk(slot_id, slot_version).unwrap();
+    if let Some(chunk) = chunk_opt.take() {
+        io::stdout().write(&chunk).unwrap();
+    }
+    process::exit(0);
+}
+
+/// Handle the get-latest-chunk subcommand
+fn handle_get_latest_chunk(mut argv: Vec<String>) {
+    let (host, contract) = parse_host_and_contract(&mut argv);
+    if argv.len() < 3 {
+        usage(Some("Expected slot_id"));
+    }
+
+    let slot_id: u32 = match argv[2].parse() {
+        Ok(x) => x,
+        Err(..) => {
+            usage(Some("Expected u32 for slot ID"));
+            unreachable!()
+        }
+    };
+
+    let mut session = StackerDBSession::new(host.clone(), contract.clone());
+    session.connect(host, contract).unwrap();
+    let chunk_opt = session.get_latest_chunk(slot_id).unwrap();
+    if let Some(chunk) = chunk_opt {
+        io::stdout().write(&chunk).unwrap();
+    }
+    process::exit(0);
+}
+
+/// Handle listing chunks
+fn handle_list_chunks(mut argv: Vec<String>) {
+    let (host, contract) = parse_host_and_contract(&mut argv);
+
+    let mut session = StackerDBSession::new(host.clone(), contract.clone());
+    session.connect(host, contract).unwrap();
+    let chunk_list = session.list_chunks().unwrap();
+    println!("{}", serde_json::to_string(&chunk_list).unwrap());
+    process::exit(0);
+}
+
+/// Handle uploading a chunk
+fn handle_put_chunk(mut argv: Vec<String>) {
+    let (host, contract) = parse_host_and_contract(&mut argv);
+    if argv.len() < 6 {
+        usage(Some("Expected slot_id, slot_version, private_key, data"));
+    }
+
+    let slot_id: u32 = match argv[2].parse() {
+        Ok(x) => x,
+        Err(..) => {
+            usage(Some("Expected u32 for slot ID"));
+            unreachable!()
+        }
+    };
+
+    let slot_version: u32 = match argv[3].parse() {
+        Ok(x) => x,
+        Err(..) => {
+            usage(Some("Expected u32 for slot version"));
+            unreachable!()
+        }
+    };
+
+    let privk = match StacksPrivateKey::from_hex(&argv[4]) {
+        Ok(x) => x,
+        Err(..) => {
+            usage(Some("Failed to parse private key"));
+            unreachable!()
+        }
+    };
+
+    let data = if argv[5] == "-" {
+        let mut buf = vec![];
+        io::stdin().read_to_end(&mut buf).unwrap();
+        buf
+    } else {
+        argv[5].as_bytes().to_vec()
+    };
+
+    let mut chunk = StackerDBChunkData::new(slot_id, slot_version, data);
+    chunk.sign(&privk).unwrap();
+
+    let mut session = StackerDBSession::new(host.clone(), contract.clone());
+    session.connect(host, contract).unwrap();
+    let chunk_ack = session.put_chunk(chunk).unwrap();
+    println!("{}", serde_json::to_string(&chunk_ack).unwrap());
+    process::exit(0);
+}
+
+fn main() {
+    let argv: Vec<String> = env::args().collect();
```

Comment:
I would prefer to parse the arguments to a concrete type instead of passing a vector of strings around to each sub-command.

Reply:
Yeah and I see clap is coming here soon https://github.com/stacks-network/stacks-blockchain/pull/3885